File: src/main/java/pl/patrykdolata/lifemanager/util/JsonUtils.java
package pl.patrykdolata.lifemanager.util;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.experimental.UtilityClass;

@UtilityClass
public class JsonUtils {

    private final ObjectMapper OBJECT_MAPPER = new ObjectMapper();

    public String asJsonString(final Object obj) {
        try {
            return OBJECT_MAPPER.writeValueAsString(obj);
        } catch (Exception e) {
            return null;
        }
    }
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/util/JsonUtils.kt
package pl.patrykdolata.lifemanager.util

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.KotlinModule

object JsonUtils {
    private val objectMapper = ObjectMapper().registerModule(KotlinModule())

    fun asJsonString(obj: Any?): String? {
        return objectMapper.writeValueAsString(obj)
    }

    fun <T> fromJson(json: String, klazz: Class<T>): T? {
        return objectMapper.readValue(json, klazz)
    }
}
